spec/openhab/core/items/semantics_spec.rb

```diff
@@ -271,4 +271,40 @@ def points(*args)
       expect(gIndoor.equipments).to match_array([Group_Equipment, NonGroup_Equipment])
     end
   end
+
+  if Semantics.respond_to?(:add) # @deprecated OH3.4 - if guard only needed with OH3.4
+    describe "#add" do
+      it "works" do
+        Semantics.add(SecretRoom: Semantics::Room)
+        expect(Semantics::SecretRoom.java_class < Semantics::Room.java_class).to be true
+
+        Semantics.add(SecretEquipment: Semantics::Equipment)
+        expect(Semantics::SecretEquipment.java_class < Semantics::Equipment.java_class).to be true
+
+        Semantics.add(SecretPoint: Semantics::Point)
+        expect(Semantics::SecretPoint.java_class < Semantics::Point.java_class).to be true
+
```

Comment:
no tests exercising the other params, nor checking the return value of #add

Reply:
I've added more tests